modules/tinymce/src/plugins/link/main/ts/core/Actions.ts

```diff
@@ -70,7 +70,12 @@ const toggleActiveState = (editor: Editor) => (api: Toolbar.ToolbarToggleButtonI
 };
 
 const toggleEnabledState = (editor: Editor) => (api: Toolbar.ToolbarButtonInstanceApi | Menu.MenuItemInstanceApi): () => void => {
-  const updateState = () => api.setEnabled(Utils.isInAnchor(editor, editor.selection.getNode()));
+  const updateState = () => {
```

Comment:
I wonder if the isInAnchor should be changed here instead so that it checks for this extra case. The original code here what pretty simple just a predicate that checks that state then sets the enabled api. If there is a risk of breaking other things maybe it could at least be a local predicate in this module or something so that the updateState function still is this simple one liner.

Reply:
`Utils.isInAnchor` is used in a several places like `toggleActiveState` above and in `ui/Controls` as well. I'm afraid that moving the selection logic there will bloat it, since we need this logic only within `toggleEnabledState` function. It's a quite specific case and shouldn't be generalized I believe.

Reply:
Ok maybe just pull it out in a local function then. That function name could then describe the thing it's doing in the ternary. Like `hasLinksInSelection` or something?

Reply:
Extracted all the code to the `hasExactlyOneLinkInSelection` function. The name is a bit verbose, but it is exactly what it does.